compiler/rustc_infer/src/infer/mod.rs

```diff
@@ -1227,23 +1227,23 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
     pub fn is_tainted_by_errors(&self) -> bool {
         debug!(
             "is_tainted_by_errors(err_count={}, err_count_on_creation={}, \
-             tainted_by_errors_flag={})",
+             tainted_by_errors={})",
             self.tcx.sess.err_count(),
             self.err_count_on_creation,
-            self.tainted_by_errors_flag.get()
+            self.tainted_by_errors.get().is_some()
         );
 
         if self.tcx.sess.err_count() > self.err_count_on_creation {
             return true; // errors reported since this infcx was made
         }
-        self.tainted_by_errors_flag.get()
+        self.tainted_by_errors.get().is_some()
     }
 
     /// Set the "tainted by errors" flag to true. We call this when we
     /// observe an error from a prior pass.
     pub fn set_tainted_by_errors(&self) {
         debug!("set_tainted_by_errors()");
-        self.tainted_by_errors_flag.set(true)
+        self.tainted_by_errors.set(Some(ErrorGuaranteed::unchecked_claim_error_was_emitted()));
```

Comment:
please use `tcx.sess.delay_span_bug` here.

Using `unchecked_claim_error_was_emitted` removes the purpose of using `ErrorEmitted` in the first place ^^

Reply:
Aha, yes!
I'm just want to aks you how to make sure the error is emitted, since ErrorGuaranteed is an empty struct.

I changed it to:
```rust
        self.tainted_by_errors.set(Some(self.tcx.sess.delay_span_bug(DUMMY_SP, "Infer type-error")));
```
I'm not sure what's a proper message show give.

Reply:
maybe something like
```rust
self.tainted_by_errors.set(Some(self.tcx.sess.delay_span_bug(DUMMY_SP, "`InferCtxt` incorrectly tainted by errors")));
```

Reply:
Done it, thank you!